linode/manager
# M3-5430: Banner for upcoming API maintenance

## Description
Adds a site-wide dismissible banner in Cloud for upcoming API maintenance events returned from Statuspage.

- [X] Fix and extend mocking for scheduled maintenances
- [X] Add `<APIMaintenanceBanner />` component
- [X] Add `apiMaintenance` feature flag & support for it
- [X] Include `<APIMaintenanceBanner />` in GlobalNotifications.tsx

To-Do:
- [x] Ensure dismissibility functions as intended in this case
- [ ] Add test (may opt to add this in a follow-up PR)

## How to test
This has to be tested in two separate situations:

a) Testing mocked Maintenances from Statuspage
1. Turn on your MSW & have the LD feature flag serve the tests
2. Observe: one site-wide banner for Linode Database Infrastructure
3. Uncomment the temp code in `useFlags.ts` to set the `apiMaintenance` flag to off/the state where there are no maintenance IDs we want to display banners for
4. Observe: the site-wide banner should not be rendered

b) Test persisted dismissibility of banner
1. Turn your MSW off & have the LD feature flag serve "Newark Maintenance"
2. Observe: a single site-wide banner for scheduled network maintenance (this is unrelated to API maintenance, but is intended as a proof-of-concept to pull an actual incident/maintenance from Statuspage into the new component and have it be dismissible)
3. Dismiss the banner. Navigating elsewhere in the app, refreshing the page, logging out and back in, etc. should not bring the banner back.
